File: mod/apphost/proto/proto.go
package proto

const (
	StatusError = "error"
	StatusOK    = "ok"
)

const (
	RequestQuery    = "connect"
	RequestRegister = "register"
)

type Request struct {
	Type     string
	Identity string
	Port     string
	Path     string
}

type Response struct {
	Status string
	Error  string
}

File: mobile/android/client/src/main/java/cc/cryptopunks/astral/proto/Proto.kt
package cc.cryptopunks.astral.proto

data class Request(
    val type: Type,
    val port: String,
    val identity: String = "",
    val path: String? = null,
) {
    enum class Type { connect, register }
}

data class Response(
    val status: String,
    val error: String = "",
) {
    enum class Status { ok, error }
}

class AstralError(error: String) : Exception(error)
